packages/react-native/Libraries/Components/Pressable/Pressable.js

```diff
@@ -340,9 +345,9 @@ function Pressable(props: Props, forwardedRef): React.Node {
       {...restPropsWithDefaults}
       {...eventHandlers}
       ref={mergedRef}
-      style={typeof style === 'function' ? style({pressed}) : style}
+      style={isStyleAFunction ? style({pressed}) : style}
       collapsable={false}>
-      {typeof children === 'function' ? children({pressed}) : children}
+      {isChildrenAFunction ? children({pressed}) : children}
```

Comment:
This has a one drawback. Let's imagine this code:
```
<Pressable onPressIn={() => setPressed(true)} onPressOut={() => setPressed(false)}>{pressed ? renderTooltip : null}</Pressable>
```
It'll break after this change, however wrong it is.
This can be overcame by just not using function children and/or function styles conditionally, but that's what we know looking at the source code. I'd suggest JSDocing something like this -> `Use function type [styles/children] to react to pressed state updates.`

Reply:
One more example, more exotic but "less wrong", is when we switch Pressables on press in, but react reuses Pressable instance under the hood. This too, however, should be covered by a simple jsdoc reminding that function styles/children should only be used when reacting to pressed change.

Reply:
Adding this to changelog might also be a good idea.

Reply:
tbh I would love to have a simplest form of a `Pressable` exposed, without such features like `styles`/`children` as functions, so we could build on top of that "pure" `Pressable` such additions if needed. Yup, PR will change the behaviour slightly and I can ofc document it if the general idea of this PR is going to be accepted. 
Maybe issue could be mitigated by storing press state in `useRef` but it's something I want to avoid.

Reply:
Definitely. I wonder if this utility is needed at all, it's really simple to implement for particular use case. Also +1 for the idea of building extra Pressable on top of this one. No idea how it could be called if stored in RN repo though.